### PR TITLE
Simplify common config args

### PR DESCRIPTION
**Pull request by JunielKatarn**
Updates the branch for #173

### CODE REVIEW

File: FFmpegConfig.sh [hunk 1]
```diff
@@ -27,6 +27,17 @@ configureArgs="\
     --enable-${sharedOrStatic} \
     --enable-cross-compile \
     --enable-debug \
+    --disable-encoders \
+    --disable-devices \
+    --disable-hwaccels \
+    --disable-programs \
+    --disable-d3d11va \
+    --disable-dxva2 \
```
Comment (lukasf) [inline FFmpegConfig.sh:35]: These should only be disabled for UWP build:

    --disable-encoders \
    --disable-devices \
    --disable-hwaccels \
    --disable-programs \
    --disable-d3d11va \
    --disable-dxva2 \

The new Win32 build is a fully featured FFmpeg.exe, so it should contain encoders, programs and all other stuff.

By the way, I think we could completely remove these:

    --disable-d3d11va \
    --disable-dxva2 \

They are both hwaccels, and we already have disable-hwaccels. So these are redundand.

**Comment by JunielKatarn** [comment]
By the way, I was able to build FFMpeg using WSL instead of MSYS2.

**Comment by lukasf** [comment]
Oh really? Removing MSYS2 would be a great simplification of the build system.

**Comment by JunielKatarn** [comment]
> Oh really? Removing MSYS2 would be a great simplification of the build system.

Yeah, except I'd need to submit a patch to ffmpeg due to file path handling (for windows, they expect a windows-style file path).

By the way, I keep getting issues when building ARM/ARM64.
Are there any specific dependencies / instructions?

Should I be using the VS-provided assemblers, or `gas-preprocessor`?

**Comment by lukasf** [comment]
Ok, dependency on an ffmpeg patch will make this more difficult, but you can try to submit a patch there. WSL does not support Windows paths?

You definitely need an up-to-date gas-preprocessor, because the ffmpeg ARM assembly code is in a format not natively understood by MS compilers.

**Comment by JunielKatarn** [comment]
I downloaded the latest gas-preprocessor today from here: https://github.com/ffmpeg/gas-preprocessor. Is there a "more official" source?

No, WSL doesn't understand Windows paths because it's a full Linux userspace.
It does have access to the Windows file system, but reads the drives as mounts (i.e. `C:` becomes `/mnt/c`).
This makes sense because in  reguilar UNIX, the colon character shouldn't be used in file paths.

**Comment by JunielKatarn** [comment]
@lukasf When you have time, could you please show me a successful ARM or ARM64 build log? (ffmpeg only).
I'm seeing the config script is giving `armasm.exe` incompatible arguments.

**Comment by lukasf** [comment]
I forgot: There is a bug in recent armasm versions that makes it fail compilation of ffmpeg sources. If you want to compile FFmpeg for ARM or ARM64, you need to install and use either 14.16 (VS2017 version) or 14.22 (last working 2019 version) C++ build tools for ARM and ARM64. You can select the version to use in our build script. An official fix is supposed to come with VS 16.7, currently in preview.

https://developercommunity.visualstudio.com/content/problem/757709/armasm-fails-to-handle-it-instructions.html

The gas-preprocessor source you used is correct.

**Comment by JunielKatarn** [comment]
> I forgot: There is a bug in recent armasm versions that makes it fail compilation of ffmpeg sources. If you want to compile FFmpeg for ARM or ARM64, you need to install and use either 14.16 (VS2017 version) or 14.22 (last working 2019 version) C++ build tools for ARM and ARM64. You can select the version to use in our build script. An official fix is supposed to come with VS 16.7, currently in preview.
> 
> https://developercommunity.visualstudio.com/content/problem/757709/armasm-fails-to-handle-it-instructions.html
> 
> The gas-preprocessor source you used is correct.

I found the issue. WSL expects a literal exe name (i.e. `armasm64.exe` instead of just `armasm`).
Changing that allowed me to build ARM variants successfully.

Note, I didn't find compilation errors using latest MSVC toolset: `14.26`.

**Comment by lukasf** [comment]
Is this a change in our config script, or does this have to be fixed in FFmpeg?

**Comment by JunielKatarn** [comment]
Our script.
I’ll update this PR reflecting that.

This does not affect using MSYS2.

**Comment by lukasf** [comment]
I just had one more idea to simplify build setup: Instead of having the devs manually download and copy gas-preprocessor (and update it when it changes), we could include that repo as a subrepository here, and add it to our path. I will try to do this in the next days.

**Comment by lukasf** [comment]
> Note, I didn't find compilation errors using latest MSVC toolset: `14.26`.

Are you sure that 14.26 was actually used? The default in the build script will select 14.1.

**Comment by JunielKatarn** [comment]
> > Note, I didn't find compilation errors using latest MSVC toolset: `14.26`.
> 
> Are you sure that 14.26 was actually used? The default in the build script will select 14.1.

Guaranteed :)
I no longer have 14.1 installed on that dev machine.
I can send you my build log, if you're curious.

I'm merging this PR today.
Looking forward for the branch to hit `master`.

**Comment by iharsh234** [comment]
> By the way, I was able to build FFMpeg using WSL instead of MSYS2.

@JunielKatarn, I am trying to build ffmpeg with nvenc inside a docker container in WSL2, curious to know how you did it with WSL?

**Comment by JunielKatarn** [comment]
> > By the way, I was able to build FFMpeg using WSL instead of MSYS2.
> 
> @JunielKatarn, I am trying to build ffmpeg with nvenc inside a docker container in WSL2, curious to know how you did it with WSL?

@iharsh234 I applied these (rather simple) patches (see attachments).

Unfortunately, they have been ignored by the FFmpeg mainstream reviewers.

[0002-Use-relative-paths-on-Windows-WSL.patch.txt](https://github.com/ffmpeginteropx/FFmpegInteropX/files/5273802/0002-Use-relative-paths-on-Windows-WSL.patch.txt)
[0001-Use-normalized-Linux-paths-on-Windows-WSL.patch.txt](https://github.com/ffmpeginteropx/FFmpegInteropX/files/5273803/0001-Use-normalized-Linux-paths-on-Windows-WSL.patch.txt)

**Pull request merged**